src/rhalphalib/sample.py

```diff
@@ -446,10 +462,14 @@ def renderRoofit(self, workspace):
                     continue
                 name = self.name + "_" + param.name + "Up"
                 shape = nominal * effect_up
+                if np.sum(shape) < 0:
+                    raise RuntimeError(f"Sample '{self.name}' has negative shape for parameter '{param.name}-Up' with value {shape}. Can't build workspace.")
                 rooTemplate = ROOT.RooDataHist(name, name, ROOT.RooArgList(rooObservable), _to_TH1(shape, self.observable.binning, self.observable.name))
                 workspace.add(rooTemplate)
                 name = self.name + "_" + param.name + "Down"
                 shape = nominal * self.getParamEffect(param, up=False)
+                if np.sum(shape) < 0:
+                    raise RuntimeError(f"Sample '{self.name}' has negative shape for parameter '{param.name}-Down with value {shape}. Can't build workspace.")
```

Comment:
There isn't a dash between param and Up/Down in the name, so I think having it here makes things confusing